app/frontend/actions.js

```diff
@@ -200,20 +198,30 @@ module.exports = ({setState, getState}) => {
 
   const verifyAddress = async (address) => {
     const state = getState()
-    if (state.usingTrezor && state.showAddressDetail) {
+    if (state.usingHwWallet && state.showAddressDetail) {
       try {
+        setState({waitingForHwWallet: true})
         await wallet.verifyAddress(state.showAddressDetail.address)
-        setState({showAddressVerification: false})
+        setState({
+          showAddressVerification: false,
```

Comment:
I wonder why do we manipulate different parts of the state
1) shouldnt' first setState alse reset `showAddressVerification` and `showAddressDetail` ?
2) why do we actually have these two parts of the state? Are they different?
3) we might prefer `finally` to clean up the state, what do you thing @refi93 ?

Reply:
**showAddressDetail** is flag to show parent modal
**showAddressVerification** is flag to show child div. If we are using hw wallet and when it's true, address and pink button with "verify on <hw-wallet>" is displayed in modal. If it's false, QR code with address and path is displayed.
**waitingForHwWallet** is defined for click feedback. When it's true, the spinner is shown on button.

Reply:
Maybe it's misleading that this state.usingHwWallet && state.showAddressDetail don't include also this `&& state.showAddressVerification`